middleware/metrics.go

```diff
@@ -0,0 +1,42 @@
+package middleware
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/trustwallet/go-libs/metrics"
+)
+
+const labelPath = "path"
+const labelMethod = "method"
+
+func MetricsMiddleware(namespace string, labels prometheus.Labels, reg prometheus.Registerer) gin.HandlerFunc {
+	perfMetric := metrics.NewPerformanceMetric(namespace, []string{labelPath, labelMethod}, labels, reg)
+	return func(c *gin.Context) {
+		path := c.FullPath()
+		method := c.Request.Method
+
+		// route not found, call next and immediately return
+		if path == "" {
```

Comment:
what is this case? 
root path is `/`?

Reply:
this happens when no route matches. `c.FullPath` returns empty string in that case

Reply:
Oh, so it's kind of 404. If route wasn't matched it just returns.